File: my-vue-app/src/api/PhongBanService.js
import axiosInstance from "./axios";
const BASE_API = "api/phongban/";
const handleError = (error) => {
  console.error("API call error:", error);
  throw error;
};
export default {
  getAllPhongBan() {
    return axiosInstance.get(`${BASE_API}getAllPhongBan`).catch(handleError);
  },

  getPhongBanID(id) {
    return axiosInstance
      .get(`${BASE_API}getPhongBanID?Id_PB=${id}`)
      .catch(handleError);
  },

  createPhongBan(data) {
    return axiosInstance
      .post(`${BASE_API}CreatePhongBan`, data)
      .catch(handleError);
  },

  updatePhongBan(data) {
    return axiosInstance
      .put(`${BASE_API}UpdatePhongBan`, data)
      .catch(handleError);
  },

  deletePhongBan(id) {
    return axiosInstance
      .delete(`${BASE_API}DeletePhongBan?id=${id}`)
      .catch(handleError);
  },
};

File: Demo_API_QL_NS/Demo_API_QL_NS/Services/PhongBan/PhongBanService.cs
using Demo_API_QL_NS.Models;
using Demo_API_QL_NS.Repositories.PhongBan;
using System.Collections;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data;
using System.Threading.Tasks;

namespace Demo_API_QL_NS.Services.PhongBan
{
    public class PhongBanService:IPhongBanService
    {
        private IPhongBanRepository _reposiory;

        public PhongBanService(IPhongBanRepository PhongBanRepository)
        {
            _reposiory = PhongBanRepository;
        }
        public async Task<IEnumerable<PhongBanDTO>> getAllPhongBan()
        {
           return await _reposiory.getAllPhongBan();

        }
        
        public async Task<bool> CreatePhongBan(PhongBanDTO model)
        {
            return await _reposiory.CreatePhongBan(model);
        }
        public async Task<bool> UpdatePhongBan(PhongBanDTO model)
        {
            return await _reposiory.UpdatePhongBan(model);
        }
        public async Task<bool> DeletePhongBan(int id)
        {
            return await _reposiory.DeletePhongBan(id);
        }
        public async Task<IEnumerable<PhongBanDTO>> SearchTenPhongBan(string Ten_PB)
        {
            return await _reposiory.SearchTenPhongBan(Ten_PB);
        }
        public async Task<PhongBanDTO> getPhongBanID(int Id_PB)
        {
            return await _reposiory.getPhongBanID(Id_PB);
        }
    }
}
